### ISSUE TITLE
Silence app crashes and loses SMS on Nexus 5X

### ISSUE THREAD

**Issue by tobia** [copperhead/bugtracker#263]
The provided SMS app "Silence" crashes as soon as it's launched, on a brand new 5X, with OS releases 2016-05-08 and 2016-05-14.

As a consequence, I lost all SMS received in the few hours until I recognized the problem, disabled it, and installed a replacement from F-Droid (QKSMS.)

Because of the SMS loss, I recommend putting a different app in the images as the default SMS handler, until this issue is fixed.

**Comment by polyzen** [comment]
I would try installing the same Silence version from F-Droid to make sure the issues are upstream. 

Please add any app recommendations to #123. :)

**Comment by tobia** [comment]
> I would try installing the same Silence version from F-Droid to make sure the issues are upstream.

Actually, my point was, whether the issue is upstream or not, it makes sense to ship a different SMS app, because on the 5X the user loses messages until they change app.

> Please add any app recommendations to #123.

I will, but #123 is about "obsolete AOSP applications", while Silence seems to be specifically chosen by Copperhead as a secure / encrypted alternative.

**Comment by polyzen** [comment]
Afaik it was simply chosen as a viable alt. to whatever SMS app may or may not be in AOSP -- the encryption just being a bonus. /shrug

**Comment by polyzen** [comment]
> How can I install the same version from F-Droid, when I cannot uninstall the stock one? (I can only disable it.)

Woops. Forgot about that. :p

**Comment by thestinger** [comment]
It works fine here. You need to provide logs. I won't be replacing it with something else.

**Comment by tobia** [comment]
What the heck is going on? I just opened it after enabling debugging options, with the intent of taking a bug report, and it didn't crash! It started and I can read the messages I thought I had lost!

I did the exact same thing before filing this bug report a few hours ago and it crashed as it had always done. This is too freaky.

**Comment by thestinger** [comment]
If you can provide logs from `adb logcat`, it might be enough to determine the problem.

**Comment by n1m1** [comment]
Same phone, same release, I am not experiencing any kind of issues related to Silence.

Cheers.

**Comment by thestinger** [comment]
Silence has been upgraded to 0.14.3 from 0.14.1. Several issues that could have led to this have been fixed upstream. Needs testing again in the next release of CopperheadOS. If other issues come up, then there's a good chance they are not the same one. Either way, I need logs at a minimum to consider app crashes as bugs.

**Comment by luther9** [comment]
I'm having this exact same problem. Build number MTC19T.2016.05.24.21.38.49. Silence version 0.14.3. I have no idea how to get logs.

**Comment by thestinger** [comment]
Via `adb logcat` with the device attached to the computer and USB debugging enabled in developer settings.

**Comment by luther9** [comment]
The Nexus 5X does not seem to have a USB port. I've never seen a charging cable like that before.

**Comment by thestinger** [comment]
It's a USB C port. Need a USB C to USB A cable.

**Comment by luther9** [comment]
Thanks, I'll see if I can get one. (I guess I'll also have to download Android SDK.)

**Comment by thestinger** [comment]
The 6P actually comes with a USB C <-> USB A cable, but LG was too cheap to include one with the 5X.

**Comment by luther9** [comment]
Ok, this is a log of me attempting to open Silence. All I can tell is that it throws a NullPointerException. Hopefully, you can read this stuff better than me. Let me know if you need any more info.

[log.txt](https://github.com/copperhead/bugtracker/files/287539/log.txt)

**Comment by thestinger** [comment]
Maybe clearing the app data in Settings -> Apps -> Silence will fix it. Could be some kind of race condition in the initial setup.

**Comment by luther9** [comment]
That did indeed fix it!

To be clear, this is what I did to fix it:

Settings > Apps > Silence > Storage > Clear Data

**Comment by thestinger** [comment]
It's possible that this is actually fixed by the new version of Silence, but it's hard to confirm it. I'm not sure what causes the problem to be triggered in the first place.

**Comment by luther9** [comment]
Well, in my case, I got the phone on Wednesday and installed both of the two most recent system updates (in chronological order; I have most recent one installed.) before trying to start Silence.

**Comment by thestinger** [comment]
I think it runs automatically as a service even before then though.

**Comment by thestinger** [comment]
There's still no known way to reproduce this, and I doubt it's a CopperheadOS bug.

**Comment by SamYaple** [comment]
I don't think this is a CopperheadOS bug either, however I can reproduce it. Powering off the system immediately after it starts on a fresh install seemed to break it for me. I have linked the related failure I had. Seems to do with it creating the initial encryption keys? Either way I was able to reproduce this 3 times in a row by flashing fresh install, rebooting as soon as android starts, before opening Silence.

Workaround for me was to simply use Silence before the first reboot. Alternatively, waiting 10 minutes before rebooting was also successful.

https://github.com/WhisperSystems/Signal-Android/issues/4155
